tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Spark.ipynb

```diff
@@ -0,0 +1,1498 @@
+   "outputs": [],
+   "source": [
+    "data_context.add_checkpoint(**checkpoint_config)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "42a4d45d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "results = data_context.run_checkpoint(\n",
+    "    checkpoint_name=\"my_checkpoint\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "10a18e22",
+   "metadata": {},
+   "source": [
+    "### Examine results by looking at DataDocs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b353ea25",
+   "metadata": {},
+   "source": [
+    "As you can see the results of the `Checkpoint` were not successful, which means the January 2020 data did not fall within the ranges predicted by the 2019 data. To examine this more closely, you can [open Data Docs here](great_expectations/uncommitted/data_docs/local_site/index.html). If the link doesn't work, from the location where this Notebook is located, you can go to `great_expectations/uncommitted/data_docs/local_site/index.html`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bfc2cc49-cd02-44db-9079-d0baed7ffe73",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "results.success"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1ef7f096-5b28-4a0a-81c1-829d57c9be23",
+   "metadata": {},
+   "source": [
+    "# Appendix"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eff94a73-740a-4a1d-b5cc-636ae6226cb5",
+   "metadata": {},
+   "source": [
+    "## What Expectation are included in the `OnboardingDataAssistant`?\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e597e3e3",
+   "metadata": {},
+   "source": [
+    "The `OnboardingDataAssistant` is built with the following `Rules` that are run internally by the `RuleBasedProfiler`. \n",
+    "\n",
+    "- `TableRule`\n",
+    "- `Column Uniqueness and Nullity Rules`\n",
+    "- `NumericColumnRule`\n",
+    "- `DateColumnRule`\n",
+    "- `TextColumnRule`\n",
+    "- `CategoricalColumnRule`\n",
+    "\n",
+    "\n",
+    "#### Table Rule\n",
+    "This Rule will take the data as a table and try to calculate the following parameter values for Expectations across batches that we pass in. \n",
+    "\n",
+    "* `expect_table_row_count_to_be_between`: \n",
+    "    - `min_value` : maximum threshold for table row count.\n",
+    "    - `max_value` : minimum threshold for table row count.\n",
+    "* `expect_table_columns_to_match_set`:\n",
+    "    - `column_set`: Either a list or set of strings, that describe the columns of the Table\n",
+    "    - `exact_match`: Boolean (default=True) which determines whether the list of columns must exactly match the observed columns. \n",
+    "    \n",
+    "#### Column Uniqueness and Nullity\n",
+    "This is not a `Rule` in the strictest sense, but the `DataAssistant` will generate `ExpectationConfigurations` for the following `Expectations` for each column in our data. \n",
```

Comment:
Yes, this is pretty deep in the code, but an attempt to be honest about the implementation :). Actually the underlying implementation has changed since this part was written. 

```markdown 
#### Rules for Uniqueness, Nullity and Non-nullity
These `Rules` each have 1 `Expectations` associated with them, and will be used by the `DataAssistant` to generate `ExpectationConfigurations` for the following `Expectations` for each column in our data. 

* `expect_column_values_to_be_unique`
* `expect_column_values_to_be_null`
* `expect_column_values_to_not_be_null`
 
They each take 2 parameters. `column`, which is the name of the column being validated, and `mostly`, which is an optional `float` value between `0` and `1` which specifies the fraction of values that match the expectation. Default for the `DataAssistant` is `1.0`. 

```